File: glsl_types.vb
Public Class GLSL_types
    
    Public GLSL_type(a As Integer) As String
        Return "int"
    End Function
    
    Public GLSL_type(a As Double) As String
        Return "double"
    End Function
    
    Public GLSL_type(a As Boolean) As String
        Return "bool"
    End Function
End Class

File: glsl_types.java
public class GLSL_types{
	public static String GLSL_type(int a){
		return "int";
	}
	public static String GLSL_type(double a){
		return "double";
	}
	public static String GLSL_type(bool a){
		return "bool";
	}
	public static String GLSL_type(bool[] a){
		return "bool["+String.valueOf(a.length)+"]";
	}
}
